frontend/src/use-cases/songs/Songs.context.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { createContext, useContext, useReducer } from "react";
+import { getSongs } from "../../api/songs/get.songs.api";
+import { getTags } from "../../api/tags/get.tags.api";
+
+export const SongTagContext = createContext({});
+
+export const SongTagProvider = ({ reducer, initialState, children }) => (
+    <SongTagContext.Provider value={useReducer(reducer, initialState)}>
+        {children}
+    </SongTagContext.Provider>
+);
+
+export const SongTagActions = {
+    LOAD_SONGS_TAGS: "LOAD_SONGS_TAGS",
+    LOAD_TAGS: "LOAD_TAGS",
+    LOADING: "LOADING"
+};
+
+export const InitialSongTagState = {
+    songs: [],
+    tags: [],
+    loadingSongs: false
+};
+
+export const SongTagReducer = (state, action) => {
+    switch (action.type) {
+        case SongTagActions.LOADING:
+            return {
+                ...state,
+                loadingSongs: true
+            };
+
+        case SongTagActions.LOAD_TAGS:
+            return {
+                ...state,
+                tags: action.tags,
+                loadingSongs: false
+            };
+
+        case SongTagActions.LOAD_SONGS_TAGS:
+            return {
+                ...state,
+                songs: action.songs,
+                tags: action.tags,
+                loadingSongs: false
+            };
+    }
+
+    return state;
+};
+
+export const useSongTag = () => {
+    const [{ songs, tags, loadingSongs }, dispatch] = useContext(
+        SongTagContext
+    );
+
+    const loadSongs = () => {
```

Comment:
This should use `useMemo`

Reply:
useMemo doesn't return a function and I can't wrap useMemo in a function either. Isn't it useCallback that I should use?

Reply:
Right yeah useCallback